### PR TITLE
[cloudevents] Add support for Tekton format of events

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Add support for events in the format used today by Tekton's own
embeded CloudEvents functionality. Control the format of events
sent via a configuration switch, that has cdevents enabled by
default and allows the Tekton format to be used for backward
compatibility.

Signed-off-by: Andrea Frittoli <andrea.frittoli@gmail.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Commit messages includes a project tag in the subject - e.g. [OCI], [hub], [results], [task-loops]

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by afrittoli** [comment]
@waveywaves FYI

File: cloudevents/pkg/apis/config/testdata/config-defaults-same.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+kind: ConfigMap
+metadata:
+  name: config-defaults
+  namespace: tekton-cloudevents
+  labels:
+    app.kubernetes.io/instance: default
+    app.kubernetes.io/part-of: tekton-cloudevents
+data:
+  default-cloud-events-sink: ""
+  default-cloud-events-format: "cdevents"
```
Comment (waveywaves) [inline cloudevents/pkg/apis/config/testdata/config-defaults-same.yaml:25]: :sparkles:

File: cloudevents/test/events/events.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	if err != nil {
+		return fmt.Errorf("error in test %s: %v", testName, err)
+	}
+	return nil
+}
+
+// eventsFromChannel takes a chan of string, a test name, and a list of events that a test
+// expects to receive. The events must be received in the same order they appear in the
+// wantEvents list. Any extra or too few received events are considered errors.
+func eventsFromChannel(c chan string, wantEvents []string) error {
```
Comment (waveywaves) [inline cloudevents/test/events/events.go:50]: :sparkles:

**Comment by waveywaves** [comment]
/approve
This looks great !

**Pull request merged**